cfgov/unprocessed/apps/filing-instruction-guide/js/fig-sidenav.js

```diff
@@ -1,68 +1,121 @@
-/* eslint-disable complexity, no-undefined */
-/* ==========================================================================
-   Scripts for Report Sidenav organism
-   ========================================================================== */
-
-const main = document.querySelector( 'main' );
-const sidenav = main.querySelector( '.o-report-sidenav' );
-const tocHeaders = sidenav.querySelectorAll( '.m-nav-link' );
-const headerOffset = main.offsetTop - 10;
-const headers = main.querySelectorAll( 'h2.report-header, h3.report-header' );
-
-const offsets = [];
-const primaryOffsets = [];
-let lastTargetIndex;
-
-// Initialize offsets for calculating what to highlight
-( function() {
-  for ( let i = 0; i < headers.length; i++ ) {
-    offsets.push( headers[i].offsetTop + headerOffset );
-    primaryOffsets.push( headers[i].tagName === 'H2' );
-  }
-} )();
+import Expandable from '@cfpb/cfpb-expandables/src/Expandable';
+import { DESKTOP, viewportIsIn } from '../../../js/modules/util/breakpoint-state.js';
+
+const appRoot = () => document.querySelector( 'main.o-fig' );
```

Comment:
Does this need to be a function/get recomputed whenever it's called? Couldn't you just cache a reference and not have to rerun the querySelector whenever you need `appRoot`?